### PR TITLE
Issue-4172 query builder preference warning text

### PR DESCRIPTION
**Pull request by carlosmbe**
Fixes Issue #4172. In this PR, I deleted the red text that was below the Query Builder's No Resreicitions Mode preference. To UX Test, make sure that "`no restctions mode`" does in fact still work and that there is no red text underneath the option in User Prefrences.

### CODE REVIEW

**Comment by carlosmbe** [comment]
# Before
Screenshot 2023-11-09 at 3 51 13 PM

# After

Screenshot 2023-11-09 at 3 41 50 PM

**Review by CarolineDenis — APPROVED** [review]

**Review by pashiav — APPROVED** [review]
Looks good!
issue-4172

**Review by melton-jason — APPROVED** [review]
:partying_face:

**Review by maxxxxxdlp — COMMENTED** [review]
Before this pull request is merged, could someone address the questions I raised in https://github.com/specify/specify7/issues/4172#issuecomment-1786286824?

TLDR: I think this warning is useful and shouldn't be removed, but I am not a user-facing person so decide what you think is best, just hear out the reasons for why this was added in the first place

**Comment by carlosmbe** [comment]
@grantfitzsimmons 
This is the comment Max is referring to:
> The reason I added it is that in no-restrictions mode you can query system tables
Then, you can open those systems tables using "Browser in Forms"
From there, it's easy to make changes or hit Delete

> If No restrictions mode is not enabled, then non-advanced users can't get to those tables (while advanced users can still change the URL to query any table they want)

> Though, now that the permission system is here, that is less fo an issue. @grantfitzsimmons if you think it is not useful, then I am ok with removing it

**Comment by grantfitzsimmons** [comment]
>TLDR: I think this warning is useful and shouldn't be removed, but I am not a user-facing person so decide what you think is best, just hear out the reasons for why this was added in the first place

I think the dangerous action is deleting the records in those tables, not the ability to query them itself. If the user doesn't have permission to delete records from those tables, there is no inherent risk. 

Maybe the text could warn against modifying records in 'advanced' tables unless the user understands the consequences.

**Comment by maxxxxxdlp** [comment]
> I think the dangerous action is deleting the records in those tables, not the ability to query them itself. If the user doesn't have permission to delete records from those tables, there is no inherent risk.

**Comment by maxxxxxdlp** [comment]
> I think the dangerous action is deleting the records in those tables, not the ability to query them itself. If the user doesn't have permission to delete records from those tables, there is no inherent risk.

I guess it depends then on how much we are relying on users there to know what they are doing before they are given the permission to delete those tables. The main collection person would have access to everything. Other collection admins would have lots of access too.

The warning on that checkbox is guarding against the case you have person with lots of power - without this query builder pref, the person with lots of power but little specify experience won't have a way to access system tables in the UI, thus there is only so much they can do to break things accidentally. with this checkbox, you don't need to know the API or do URL editing to be able to delete from various tables - way lower bar.

anyway, just my 2c. feel free to soften the warning or remove it
